includes/namespace.php

```diff
@@ -29,63 +29,117 @@
 use WP_Error;
 use WP_Site;
 
-if (
-	! class_exists( '\Google\Web_Stories\Plugin' ) ||
-	! file_exists( WEBSTORIES_PLUGIN_DIR_PATH . '/assets/js/edit-story.js' )
-) {
-	/**
-	 * Displays an admin notice about why the plugin is unable to load.
-	 *
-	 * @since 1.0.0
-	 *
-	 * @return void
-	 */
-	function _print_missing_build_admin_notice() {
-		?>
-		<div class="notice notice-error">
-			<p>
-				<strong><?php esc_html_e( 'Web Stories plugin could not be initialized.', 'web-stories' ); ?></strong>
-			</p>
-			<p>
-				<?php
-					echo wp_kses(
-						sprintf(
-						/* translators: %s: build commands. */
-							__( 'You appear to be running an incomplete version of the plugin. Please run %s to finish installation.', 'web-stories' ),
-							'<code>composer install &amp;&amp; npm install &amp;&amp; npm run build</code>'
-						),
-						[
-							'code' => [],
-						]
-					);
-				?>
-			</p>
-		</div>
-		<?php
-	}
+// Load Compatibility class the old fashioned way.
+if ( ! class_exists( '\Google\Web_Stories\Compatibility' ) ) {
+	require_once WEBSTORIES_PLUGIN_DIR_PATH . 'includes/Compatibility.php';
+}
+
+global $web_stories_compatibility;
+
+$web_stories_error = new WP_Error();
+$extensions        = [
+	'date'   => [
+		'classes' => [
+			'DateTimeImmutable',
+		],
+	],
+	'dom'    => [
+		'classes' => [
+			'DOMAttr',
+			'DOMComment',
+			'DOMDocument',
+			'DOMElement',
+			'DOMNode',
+			'DOMNodeList',
+			'DOMText',
+			'DOMXPath',
+		],
+	],
+	'json'   => [
+		'functions' => [
+			'json_decode',
+			'json_encode',
+		],
+	],
+	'libxml' => [
+		'functions' => [
+			'libxml_use_internal_errors',
+		],
+	],
+	'spl'    => [
+		'functions' => [
+			'spl_autoload_register',
+		],
+	],
+];
+
+$web_stories_compatibility = new Compatibility( $web_stories_error );
+$web_stories_compatibility->set_extensions( $extensions );
+$web_stories_compatibility->set_php_version( WEBSTORIES_MINIMUM_PHP_VERSION );
+$web_stories_compatibility->set_wp_version( WEBSTORIES_MINIMUM_WP_VERSION );
+$web_stories_compatibility->set_required_files(
+	[
+		WEBSTORIES_PLUGIN_DIR_PATH . '/assets/js/edit-story.js',
+		WEBSTORIES_PLUGIN_DIR_PATH . '/assets/js/stories-dashboard.js',
+		WEBSTORIES_PLUGIN_DIR_PATH . '/assets/js/web-stories-embed-block.js',
```

Comment:
```suggestion
		WEBSTORIES_PLUGIN_DIR_PATH . '/assets/js/web-stories-embed-block.js',
		WEBSTORIES_PLUGIN_DIR_PATH . '/third-party/vendor/scoper-autoload.php',
		WEBSTORIES_PLUGIN_DIR_PATH . '/includes/vendor/autoload.php',
```

@swissspidy Thoughts on check these files as well?

Reply:
Sounds good, but then `set_class_name` and the class name checks are superfluous.

If we know these files exist and `spl_autoload_register` exists there's no way the class would not be available.

Reply:
Okay, I will just keep `/third-party/vendor/scoper-autoload.php'` check in for now.